src/prefect/transactions.py

```diff
@@ -195,7 +195,9 @@ def __enter__(self):
             and not self.store.supports_isolation_level(self.isolation_level)
         ):
             raise ValueError(
-                f"Isolation level {self.isolation_level.name} is not supported by provided result store."
+                f"Isolation level {self.isolation_level.name} is not supported by provided "
```

Comment:
note for future: I think we should refactor many of these errors into special `prefect.exceptions.ConfigurationError` exceptions for clarity and consistency